providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/config.py

```diff
@@ -0,0 +1,33 @@
+import os
+from dataclasses import dataclass, field
+
+
+def str_to_bool(val: str) -> bool:
+    return val.lower() == "true"
+
+
+def config(env_var, default, cast=None):
+    """
+    Implements a field that falls back to an environment variable or a
+    default value if a value is not provided.
+    """
+
+    def env_or_default(env_var, default, cast=None):
+        val = os.environ.get(env_var)
+        if val is None:
+            return default
+        return val if cast is None else cast(val)
+
+    return field(default_factory=lambda: env_or_default(env_var, default, cast=cast))
+
+
+@dataclass
+class Config:
+    host: str = config("FLAGD_HOST", "localhost")
+    port: str = config("FLAGD_PORT", 8013, cast=int)
+    tls: bool = config("FLAGD_TLS", False, cast=str_to_bool)
+    timeout: int = 5
+
+    @classmethod
+    def build(cls, **kwargs):
+        return cls(**{k: v for k, v in kwargs.items() if v is not None})
```

Comment:
why not just use a normal class, then you don't need the `build()` method and you could remove the `config()` function and just use the inner function `env_or_default()` directly
```suggestion
class Config:
    def __init__(
        self,
        host: typing.Optional[str] = None,
        port: typing.Optional[int] = None,
        tls: typing.Optional[bool] = None,
        timeout: typing.Optional[int] = None,
    ):
        self.host = env_or_default("FLAGD_HOST", "localhost") if host is None else host
        self.port = env_or_default("FLAGD_PORT", 8013, cast=int) if port is None else port
        self.tls = env_or_default("FLAGD_TLS", False, cast=str_to_bool) if tls is None else tls
        self.timeout = 5 if timeout is None else timeout
```

Reply:
Yeah, I might have been a little too clever for my own good here, and got carried away.